Assets/Scripts/WeaponStation.cs

```diff
@@ -48,17 +48,25 @@ void Start()
     }
 
     public override void Interact(){
-        if (this.weaponOnSale == true){
-            player.PickupWeapon(this.weapon);
-            this.weaponOnSale = false;
-            this.restockInProgress = true;
-            animator.SetBool("Reload", true);
-        }
-        else if (this.weaponOnSale == false & this.restockInProgress == false){
-            player.RefillAmmoFromStation(this.weapon);
-            this.restockInProgress = true;
-            animator.SetBool("Reload", true);
-            this.ammoSprite.SetActive(false);
+        if (this.CheckIfEnoughMoney() == true){
+            if (this.weaponOnSale == true){
+                this.ChangeMoney();
+                player.PickupWeapon(this.weapon);
+                this.weaponOnSale = false;
+                this.restockInProgress = true;
+                animator.SetBool("Reload", true);
+                this.cost = ammoCost;
```

Comment:
I don't think you ever change the cost back after it is set to `ammoCost` so once the weapon has been purchased the first time the cost will never go back to the original - I think it's probably worth having a discussion about prices any (we might just keep the same price for both)